File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/iban/ui/IBANPresenterModule.java
package com.fintechplatform.ui.iban.ui;


import com.fintechplatform.api.enterprise.api.EnterpriseAPI;
import com.fintechplatform.api.iban.api.IbanAPI;
import com.fintechplatform.api.profile.api.ProfileAPI;
import com.fintechplatform.ui.enterprise.db.enterprise.EnterprisePersistanceDB;
import com.fintechplatform.ui.iban.db.IbanPersistanceDB;
import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class IBANPresenterModule {

    private IBANContract.View view;
    private DataAccount configuration;

    public IBANPresenterModule(/*IBANContract.View view,*/ DataAccount configuration){
        //this.view = view;
        this.configuration = configuration;
    }

    @Singleton
    @Provides
    IBANContract.Presenter providesIBanPresenter(//IBANContract.View view,
                                                 IbanAPI api,
                                                 ProfileAPI profileAPI,
                                                 EnterpriseAPI enterpriseAPI,
                                                 IbanPersistanceDB ibanPersistanceDB,
                                                 UsersPersistanceDB usersPersistanceDB,
                                                 EnterprisePersistanceDB enterprisePersistanceDB){
        return new IBANPresenter(api, profileAPI, enterpriseAPI, configuration, ibanPersistanceDB, usersPersistanceDB, enterprisePersistanceDB);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/iban/di/IBANPresenterModule.kt
package com.fintechplatform.ui.iban.di

import com.fintechplatform.api.enterprise.api.EnterpriseAPI
import com.fintechplatform.api.iban.api.IbanAPI
import com.fintechplatform.api.profile.api.ProfileAPI
import com.fintechplatform.ui.enterprise.db.enterprise.EnterprisePersistanceDB
import com.fintechplatform.ui.iban.IBANContract
import com.fintechplatform.ui.iban.IBANPresenter
import com.fintechplatform.ui.iban.db.IbanPersistanceDB
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class IBANPresenterModule(private val view: IBANContract.View, private val configuration: DataAccount) {

    @Singleton
    @Provides
    internal fun providesIBanPresenter(api: IbanAPI,
                                       profileAPI: ProfileAPI,
                                       enterpriseAPI: EnterpriseAPI,
                                       ibanPersistanceDB: IbanPersistanceDB,
                                       usersPersistanceDB: UsersPersistanceDB,
                                       enterprisePersistanceDB: EnterprisePersistanceDB): IBANContract.Presenter = 
            IBANPresenter(view, api, profileAPI, enterpriseAPI, configuration, ibanPersistanceDB, usersPersistanceDB, enterprisePersistanceDB)
}